### PR TITLE
Upgrade to check-spelling v0.0.22

### PR DESCRIPTION
**Pull request by jsoref**
- [x] I have signed the [Contributor License Agreement](https://cla.opensource.microsoft.com/microsoft/winget-pkgs).
- [x] This pull request is related to https://github.com/microsoft/winget-cli/pull/3885#issuecomment-1809439464

-----

Changes:
* general spelling/grammar
* refreshed baseline metadata (roughly [spell-check-this@edccef0](https://github.com/check-spelling/spell-check-this/tree/edccef0d97da7e227dbec1d2c1faac5fd085f313))
* upgraded to [check-spelling v0.0.22](https://github.com/check-spelling/check-spelling/releases/tag/v0.0.22)
   * ⚠️ there's one pattern that triggers in v0.0.21 (as part of this PR) which will not trigger once this PR is merged ([🩹 Pattern suggestions shouldn't trip over active patterns](https://github.com/check-spelling/check-spelling/wiki/Feature:-Suggest-patterns#multiple-patterns-for-a-line))
* refreshed workflow
  * slightly refreshed dictionaries -- I'm removing `cspell:cpp/src/cpp.txt`, but you could use it or `cspell:cpp/dict/cpp.txt` -- I think I missed that #3885 was adding it -- cpp suffers from the same problem as win32 -- too many misspelled words)
  * dropping commenting on pushes (people should migrate to github step summaries)

 ###### Microsoft Reviewers: [Open in CodeFlow](https://microsoft.github.io/open-pr/?codeflow=https://github.com/microsoft/winget-cli/pull/3896)

### CODE REVIEW

File: .github/actions/spelling/excludes.txt
Comment (jsoref) [inline .github/actions/spelling/excludes.txt:93]: afaict this is no longer a problem, so I'm dropping it

File: .github/actions/spelling/expect.txt
Comment (jsoref) [inline .github/actions/spelling/expect.txt:430]: These `T[A-Z]` items are handled by a pattern below.

File: .github/actions/spelling/patterns.txt
Comment (jsoref) [inline .github/actions/spelling/patterns.txt:72]: There were a handful of `T...` items (flagged earlier), and it seemed worthwhile to add `T` here. There are also a couple of other letters which each have a couple of items, but it didn't really seem worthwhile to include a blanket -- it is worth noting that the pattern could be widened to cover them if that's desired.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:86]: this is required for sarif reporting

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:90]: this is purely stylistic -- the vscode github integration doesn't like `"..."` whereas GitHub Actions do not mind.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:100]: This will let dependabot PRs warn about when changing the workflow will break things, especially `cspell:...` items.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:103]: this is mostly irrelevant at this point (it's used when a branch doesn't have check-spelling metadata -- which basically happens when it's first added to a repository). I suppose the alternative would be to remove the lines entirely.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:109]: The change from 10 to 20 is something I'm toying with in spell-check-this, but it might not make sense here.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:84]: this dictionary is lousy (way too many typos included, similar to the win32 dictionary) -- as of v0.0.22, I [🗑️ Removed (both) Dictionaries](https://github.com/check-spelling/check-spelling/wiki/Removed-Dictionaries) from the suggestion list due to their poor quality.

I did look into trying to fix them, but they had so many typos that I ran out of energy trying to go over them.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:117]: order doesn't matter, so I should probably shift things so that they aren't changing in this PR.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:178]: This is, as noted, very optional, but it should make it easier for others to interact with the workflow instead of having to run programs on their own computer.

File: .github/actions/spelling/candidate.patterns
Comment (florelis) [inline .github/actions/spelling/candidate.patterns:1]: So much of this file seems to be specific to technologies we don't use in the project. I'm guessing we keep it all because it's easier to update if we don't make changes to the standard file?

File: .github/actions/spelling/excludes.txt
Comment (florelis) [inline .github/actions/spelling/excludes.txt:106]: This seems duplicated

File: .github/actions/spelling/excludes.txt
Comment (florelis) [inline .github/actions/spelling/excludes.txt:123]: Could this be a single line up to ExternalModules to exclude the whole directory?

File: .github/actions/spelling/patterns.txt
Comment (florelis) [reply .github/actions/spelling/patterns.txt:72]: Does this still check the spelling of the word after the `T`? Or would it allow stuff like `TMispeled`?

File: .github/actions/spelling/expect.txt
Comment (florelis) [inline .github/actions/spelling/expect.txt:581]: Nit: can these be sorted?

File: .github/workflows/spelling3.yml
Comment (florelis) [reply .github/workflows/spelling3.yml:84]: I can see why you wouldn't want to suggest the lousy dictionaries, but I'm not totally convinced it is better for us to not use them. I'm not too invested in that position, though.

The downside I see to using them is that if we happen to make the same typo, we wouldn't notice. To me that seems somewhat unlikely, and if it does happen doesn't seem like that big of a deal. I assume the dictionary would get better over time, so those typos would eventually come up.
As for upsides, it will be easier for us to manage the lists of allow/expect if we have to add less words. If we have to add lots of words, I think it's more likely we're going to make another mistake like with "uknown"

File: .github/workflows/spelling3.yml
Comment (florelis) [inline .github/workflows/spelling3.yml:77]: What does this do? Check spelling when someone creates an issue?

**Review by florelis — APPROVED** [review]

File: .github/actions/spelling/candidate.patterns
Comment (jsoref) [reply .github/actions/spelling/candidate.patterns:1]: Yep.

File: .github/actions/spelling/excludes.txt
Comment (jsoref) [reply .github/actions/spelling/excludes.txt:123]: Yes

File: .github/actions/spelling/expect.txt
Comment (jsoref) [reply .github/actions/spelling/expect.txt:581]: These will go away once it merges, they should be covered by one of the other dictionaries or some other part of the upgrade.

File: .github/workflows/spelling3.yml
Comment (jsoref) [reply .github/workflows/spelling3.yml:77]: When someone comments on an issue/PR it will trigger the workflow. Each job independently decides whether to do anything.

For,
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L80
It'll do nothing:
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L90

For,
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L121
It'll do nothing:
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L128

For,
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L136
It'll do nothing:
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L144

It will only do work here:
https://github.com/microsoft/winget-cli/blob/64d4149c1cf690f66c6a01cfaa9d1eb2fba18ac3/.github/workflows/spelling3.yml#L153-L165
One of the criteria being that it isn't this repository.

If someone creates a PR within a single fork and it triggers a comment w/ a note about items that should be added to expect, the comment will include something like this from @check-spelling-bot:

File: .github/actions/spelling/patterns.txt
Comment (jsoref) [reply .github/actions/spelling/patterns.txt:72]: Yes, the `(?=...)` is a [zero-width lookahead](https://www.regular-expressions.info/lookaround.html)

File: .github/workflows/spelling3.yml
Comment (JohnMcPMS) [reply .github/workflows/spelling3.yml:86]: I think I have to veto on this permission.  I don't know what feature this removes, but allowing an action outside of our control to write is not going to work.  If they had `create` or similar permission level so that you could create new items but not overwrite existing ones, then it would likely be more acceptable.

**Review by JohnMcPMS — CHANGES_REQUESTED** [review]
Cannot allow write permissions.

File: .github/workflows/spelling3.yml
Comment (jsoref) [reply .github/workflows/spelling3.yml:86]: Yeah, I agree. I'm really annoyed that there isn't such a thing. If you can find someone to poke to get them to make a better api surface, I'd be overjoyed 😉.

Comment (JohnMcPMS) [reply .github/workflows/spelling3.yml:86]: [community/discussions#29710](https://github.com/orgs/community/discussions/29710)

Closest thing I found.

**Comment by denelon** [comment]
We use a "template: foo/bar/baz" type of convention for messages from bots so it's easy to build e-mail rules to filter those out. How hard would it be to add something like that to the spellcheck messages in PRs?

**Comment by jsoref** [comment]
@denelon, add a line to https://github.com/microsoft/winget-cli/blob/master/.github/actions/spelling/advice.md

You could even add it of the form:
```
<!--
template: check-spelling
-->
```

So that it's only seen by mail filters and not by humans.

File: src/WinGetUtilInterop.UnitTests/TestCollateral/AllEquality.yaml
Comment (jsoref) [inline src/WinGetUtilInterop.UnitTests/TestCollateral/AllEquality.yaml:24]: I think I'd been ignoring this for a while... contoso.com is microsoft's domain. contosa is a typo of contoso, and contosa.net isn't registered.

If this needs to _not_ be contoso.com, it should be an example.com or some other microsoft controlled domain...

File: .github/actions/spelling/advice.md
Comment (jsoref) [reply .github/actions/spelling/advice.md:29]: @denelon I hope this meets your needs, but you're welcome to change it to some other pattern.

**Comment by jsoref** [comment]
Hmm.. I guess I need to rebase -- I was hoping @denelon or @JohnMcPMS would have reviewed it since...

At this point, I'm close to releasing the next version...

**Comment by denelon** [comment]
Sorry about that! We've been super busy with some internal work and haven't been as good as we should have been about reviewing external PRs. Do you want me to get someone to review this or hold off until the next release?

File: .github/workflows/spelling3.yml
Comment (jsoref) [reply .github/workflows/spelling3.yml]: I'm dropping this for now.

File: .github/workflows/spelling3.yml
Comment (jsoref) [inline .github/workflows/spelling3.yml:98]: I'm removing this job from the flow.

In the long term, I'm nudging people to rely on the step summary (which happens automatically) over adding comments, but in the near term, people can rely on comments when they make a pull request.

**Review by florelis — APPROVED** [review]

**Review by JohnMcPMS — APPROVED** [review]

**Comment by jsoref** [comment]
@JohnMcPMS  could you add `check-spelling/actions-checkout@*` to the exemptions list? It isn't used at this time, but because of the logic, the runtime doesn't know.

```
Check Spelling
Bad request - check-spelling/actions-checkout@v4 is not allowed to be used in microsoft/winget-cli. Actions in this workflow must be: within a repository that belongs to your Enterprise account, created by GitHub, or matching the following: check-spelling/check-spelling@*, check-spelling/checkout-merge@*, craigloewen-msft/gitgudsimilarissues@*.
```

**Comment by denelon** [comment]
@jsoref, I just added it.

**Pull request merged**
